src/taskpane/pages/citationStyle.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from "react";
+import { FocusZone, FocusZoneDirection } from "@fluentui/react/lib/FocusZone";
+import { List } from "@fluentui/react/lib/List";
+import {
+  ITheme,
+  mergeStyleSets,
+  getTheme,
+  getFocusStyle,
+} from "@fluentui/react/lib/Styling";
+import { Stack } from "@fluentui/react";
+
+const theme: ITheme = getTheme();
+const { palette, semanticColors, fonts } = theme;
+const classNames = mergeStyleSets({
+  container: {
+    overflow: "auto",
+    padding: "0.25rem 0.25rem 0px",
+    webkitBoxFlex: "1 1 auto",
+  },
+  itemCell: [
+    getFocusStyle(theme, { inset: -1 }),
+    {
+      minHeight: 34,
+      boxSizing: "border-box",
+      padding: "0 25px",
+      display: "flex",
+      selectors: {
+        "&:hover": { background: palette.themeLighterAlt },
+      },
+    },
+  ],
+  itemName: [
+    fonts.medium,
+    {
+      padding: "8px 0",
+      borderBottom: `1px solid ${semanticColors.bodyDivider}`,
+      minWidth: "100%",
+    },
+  ],
+  StyleHeading: [
+    fonts.medium,
+    {
+      color: palette.neutralSecondary,
+      fontWeight: "bold",
+      padding: 20,
+      paddingBottom: 5,
+      paddingTop: 15,
+    },
+  ],
+  selectedStyle: [
+    fonts.medium,
+    {
+      color: palette.themeDarkAlt,
+      padding: 20,
+      paddingBottom: 5,
+      paddingTop: 5,
+      fontWeight: "bold",
+    },
+  ],
+});
+
+function CitationStyle(): JSX.Element {
+  const items = [
+    {
+      text: "American Political Science Association",
+      value: "american-political-science-association",
+    },
+    { text: "IEEE", value: "ieee" },
+    {
+      text: "American Sociological Association 6th edition",
+      value: "american-sociological-association",
+    },
+    {
+      text: "American Psychological Association 7th edition",
+      value: "advances-in-complex-systems",
+    },
+    {
+      text: "Chicago Manual of Style 16th edition (author-date)",
+      value: "chicago-author-date-16th-edition",
+    },
+  ];
+  const preferenceStyle: string | null = Office.context.document.settings.get(
+    "Style"
+  ) as string;
+  const selectedStyle = preferenceStyle
+    ? items.find((item) => item.value === preferenceStyle).text
```

Comment:
Is there a reason not to use the `value` instead of `text` in the preferences? It's currently inconsistent with the component state that indeed uses the `value`.

Reply:
>  It's currently inconsistent with the component state that indeed uses the value.

Right, But the value(filename) is being saved in the preferences. Here in this line, we are getting the text from the items array corresponding to the value we get from the saved preferences to set up the initial state.

The preferenceStyle is the value(filename) of the style.

Reply:
Right, but why do you need to know the text for the style value in the preferences? If the state (in the preferences and here in the react component) is always the file name, then you never need to access the text of the style, except for displaying purposes, right? This would simplify a few things, for example below you can just call `setCurrentStyle(ev.currentTarget.id)` instead of needing to transverse all items to find the matching one, and then store the corresponding text.

Reply:
I agree. We can avoid one traversal in the onClick function. But here, after the **first render,** we get the value(filename) from the preferences, and we need the text corresponding to that value to display. Maybe I'm missing something. Let's discuss this in the meeting.

Reply:
My point was to have the state returned by `React.useState` to be still the value (filename), and then put the conversion filename -> display text below directly in the render method.